### PR TITLE
Remove tests for .NET SDK 3.1 and 5, add tests for .NET 9

### PR DESCRIPTION
**Pull request by zijchen**
Remove EOL .NET SDK versions and add .NET 9

### CODE REVIEW

File: .github/workflows/pr-validation.yml [hunk 1]
```diff
@@ -9,23 +9,26 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
-        dotnetVersion: [3.1.x, 5.x, 6.x, 7.x, 8.x]
+        dotnetVersion: [6.x, 7.x, 8.x, 9.x]
```
Comment (ErikEJ) [inline .github/workflows/pr-validation.yml:12]: Why are you adding .NET 7, it is out of support?

Comment (zijchen) [reply .github/workflows/pr-validation.yml:12]: .NET 7 is still supported so we'll keep it there for now

Comment (ErikEJ) [reply .github/workflows/pr-validation.yml:12]: I dont think so: https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

Comment (zijchen) [reply .github/workflows/pr-validation.yml]: Oh wow looks like it just ended yesterday, although I don't think there's any harm in testing for NET 7 too for now.

Comment (llali) [reply .github/workflows/pr-validation.yml]: why not removing net7 if it's out of support?

**Review by llali — APPROVED** [review]

**Pull request merged**